use of deprecated beautifulsoup method findAll causes tests to fail

When building 3.21.0 for alpine linux some of the tests fail as the deprecation warning is treated as an error:

```
_ EasynameProviderTests.test_provider_when_calling_create_record_for_A_with_valid_name_and_content _
self = <test_easyname.EasynameProviderTests testMethod=test_provider_when_calling_create_record_for_A_with_valid_name_and_content>
>   ???
tests/providers/integration_tests.py:131: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.testenv/lib/python3.12/site-packages/lexicon/_private/providers/easyname.py:106: in create_record
    ???
.testenv/lib/python3.12/site-packages/lexicon/_private/providers/easyname.py:126: in _create_record_internal
    ???
.testenv/lib/python3.12/site-packages/lexicon/_private/providers/easyname.py:336: in _is_duplicate_record
    ???
.testenv/lib/python3.12/site-packages/lexicon/_private/providers/easyname.py:229: in list_records
    ???
.testenv/lib/python3.12/site-packages/lexicon/_private/providers/easyname.py:259: in _list_records_internal
    ???
.testenv/lib/python3.12/site-packages/lexicon/_private/providers/easyname.py:370: in _get_dns_entry_trs
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <table style="width: 100%;">
<thead>
<tr class="entity--dns-record entity__header-row">
<th class="entity__field entit...2 15l-4 1 1-4 9.5-9.5z"></path></svg>
</span>
<span class="sr-only">Edit the DNS entry</span>
</a>
</td>
</tr>
</table>
args = (<function Provider._get_dns_entry_trs.<locals>._is_zone_tr at 0x7f4818440220>,)
kwargs = {}
>   ???
E   DeprecationWarning: Call to deprecated method findAll. (Replaced by find_all) -- Deprecated since version 4.0.0.
/usr/lib/python3.12/site-packages/bs4/_deprecation.py:56: DeprecationWarning
```

Full build log here: https://gitlab.alpinelinux.org/a16bitsysop/aports/-/jobs/1814446

--- a/src/lexicon/_private/providers/easyname.py
+++ b/src/lexicon/_private/providers/easyname.py
@@ -273,7 +273,7 @@ class Provider(BaseProvider):
                     rec["priority"] = (columns[3].contents[1].string or "").strip()
                     rec["ttl"] = (columns[4].contents[1].string or "").strip()
                     rec["id"] = ""
-                    for a in columns[5].findAll(
+                    for a in columns[5].find_all(
                         "a", class_="button button--transparent"
                     ):
                         rec["id"] = int(a["href"].rsplit("/", 1)[-1])
@@ -367,7 +367,7 @@ class Provider(BaseProvider):
         def _is_zone_tr(elm):
             return elm.name.lower() == "tr" and (elm.has_attr("class"))
 
-        rows = dns_table.findAll(_is_zone_tr)
+        rows = dns_table.find_all(_is_zone_tr)
         assert rows is not None and rows, "Could not find any DNS entries"
         return rows
 

--- a/src/lexicon/_private/providers/henet.py
+++ b/src/lexicon/_private/providers/henet.py
@@ -144,7 +144,7 @@ class Provider(BaseProvider):
         def is_dns_tr_type(klass):
             return klass and re.compile("dns_tr").search(klass)
 
-        records = html.findAll("tr", class_=is_dns_tr_type)
+        records = html.find_all("tr", class_=is_dns_tr_type)
 
         # If the tag couldn't be found, error, otherwise, return the value of the tag
         if records is None or not records:
@@ -153,7 +153,7 @@ class Provider(BaseProvider):
 
         new_records = []
         for dns_tr in records:
-            tds = dns_tr.findAll("td")
+            tds = dns_tr.find_all("td")
             # Process HTML in the TR children to derive each object
             rec = {}
             rec["zone_id"] = tds[0].string

